cogs/perpetuum_killboard_cog.py

```diff
@@ -101,52 +158,88 @@ async def display_new_kills(self):
-                    attack_text += ", {} ECM".format(
-                        a['totalEcmAttempts'],
-                    )
-                if int(a['sensorSuppressions']) > 0:
-                    attack_text += ", {} SS".format(
-                        a['sensorSuppressions'],
-                    )
-
-                if float(a['energyDispersed']) > 0:
-                    attack_text += ", {} Accum drained".format(
-                        a['energyDispersed'],
-                    )
 
+            # Embed Setup
+            kill_message_embed = discord.Embed(title="Killboard Link",
+                                               url="https://killboard.openperpetuum.com/kill/" + str(kill['id']),
+                                               color=discord.colour.Color.random())
+            kill_message_embed.set_author(name="Killmail #" + str(kill['id']),
+                                          url="https://api.openperpetuum.com/killboard/kill/" + str(kill['id']))
+
+            kill_message_embed.set_footer(text="Happened on " + str(kill['date']))
+
+            # Embed - Target
+            kill_message_embed.add_field(name="Target",
+                                         value=str("🕵️ **Agent**: " + str(kill['_embedded']['agent']['name'])) +
+                                               "\n💠 **Corp**: " + str(kill['_embedded']['corporation']['name']) +
+                                               "\n🤖 **Robot**: " + bot_name_lookup.get(
+                                             kill['_embedded']['robot']['definition']) +
+                                               "\n🩹 **Damage Taken**: " + prettier_numbers(kill['damageReceived']) +
+                                               "\n🗺️ **Zone**: " + str(
+                                             kill['_embedded']['zone']['name']) + embed_linebreak(),
+                                         inline=False)
+
+            # Embed - Attacker(s) | Discord has some limitations when it comes to Embeds
+            # More info: https://discordjs.guide/popular-topics/embeds.html#embed-limits
+
+            # Setup for Attacker Field(s) for Embed
+            atk_field_name = "⚔ Attackers"
+            atkcount = (str(len(kill['_embedded']['attackers'])))
+            if int(atkcount) <= 1:  # If there's only 1 attacker, change field name to singular.
+                atk_field_name = "⚔ Attacker"
+
+            atk_saved_list = ""  # Validated Attackers, saved to a list - These WILL exist in the embed
+            current_atk_str = ""  # Validated Attackers + New Attacker we want to add - Check if this exceeds limits
+            current_field_length = 0  # char-length of field we're currently manipulating
+            embed_length = len(kill_message_embed)  # Embed char length, MAX 6000
+            fields_built = len(kill_message_embed.fields)  # Keep track of total fields, MAX 25
+            inline_next_attacker_field = False
```

Comment:
I thought it was more readable to understand the voodoo later, if these were initialized early and set to empty values with comments.
Technically not all are needed.